ui/button.js

```diff
@@ -56,355 +39,267 @@ $.widget( "ui.button", {
 		label: null,
-		icons: {
-			primary: null,
-			secondary: null
+		icon: null,
+		iconPosition: "beginning",
+		classes: {
+			"ui-button": "ui-corner-all"
 		}
 	},
-	_create: function() {
-		this.element.closest( "form" )
-			.unbind( "reset" + this.eventNamespace )
-			.bind( "reset" + this.eventNamespace, formResetHandler );
 
-		if ( typeof this.options.disabled !== "boolean" ) {
-			this.options.disabled = !!this.element.prop( "disabled" );
-		} else {
-			this.element.prop( "disabled", this.options.disabled );
-		}
+	_getCreateOptions: function() {
+		var disabled,
+			options =  this._super() || {};
 
-		this._determineButtonType();
-		this.hasTitle = !!this.buttonElement.attr( "title" );
+		this.isInput = this.element.is( "input" );
+		this.originalLabel = this.isInput ? this.element.val() : this.element.html();
 
-		var that = this,
-			options = this.options,
-			toggleButton = this.type === "checkbox" || this.type === "radio",
-			activeClass = !toggleButton ? "ui-state-active" : "";
+		disabled = this.element[ 0 ].disabled;
+		if ( disabled != null ) {
+			options.disabled = disabled;
+		}
 
-		if ( options.label === null ) {
-			options.label = (this.type === "input" ? this.buttonElement.val() : this.buttonElement.html());
+		if ( this.originalLabel ) {
+			options.label = this.originalLabel;
 		}
 
-		this._hoverable( this.buttonElement );
+		return options;
+	},
 
-		this.buttonElement
-			.addClass( baseClasses )
-			.attr( "role", "button" )
-			.bind( "mouseenter" + this.eventNamespace, function() {
-				if ( options.disabled ) {
-					return;
-				}
-				if ( this === lastActive ) {
-					$( this ).addClass( "ui-state-active" );
-				}
-			})
-			.bind( "mouseleave" + this.eventNamespace, function() {
-				if ( options.disabled ) {
-					return;
-				}
-				$( this ).removeClass( activeClass );
-			})
-			.bind( "click" + this.eventNamespace, function( event ) {
-				if ( options.disabled ) {
-					event.preventDefault();
-					event.stopImmediatePropagation();
-				}
-			});
+	_create: function() {
+		var formElement = $( this.element[ 0 ].form );
 
-		// Can't use _focusable() because the element that receives focus
-		// and the element that gets the ui-state-focus class are different
-		this._on({
-			focus: function() {
-				this.buttonElement.addClass( "ui-state-focus" );
-			},
-			blur: function() {
-				this.buttonElement.removeClass( "ui-state-focus" );
-			}
-		});
+		// We don't use _on and _off here because we want all the checkboxes in the same form to use
+		// single handler which handles all the checkboxradio widgets in the form
+		formElement.off( "reset" + this.eventNamespace, formResetHandler );
+		formElement.on( "reset" + this.eventNamespace, formResetHandler );
 
-		if ( toggleButton ) {
-			this.element.bind( "change" + this.eventNamespace, function() {
-				that.refresh();
-			});
+		if ( this.options.disabled == null ) {
+			this.options.disabled = this.element[ 0 ].disabled || false;
```

Comment:
We've been over this at least once, but since there are no comments and I can't remember the details, I have to ask again: Why is this also checking `this.element[ 0 ].disabled`, when `_getCreateOptions` already does that?